src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -130,9 +130,31 @@ define(function (require, exports, module) {
     }
 
     // Get Analytics data
-    function getAnalyticsData() {
+    function getAnalyticsData(EventParams) {
         var userUuid = PreferencesManager.getViewState("UUID"),
             olderUuid = PreferencesManager.getViewState("OlderUUID");
+        var eventParams = {};
+
+        if (!EventParams || EventParams === undefined) {
```

Comment:
This is actually not correct.
In JS !obj will return true, in all cases where obj is false, null or undefined, so the second check is not required.

Reply:
This can be much easily done in the following way:
Have a default parameter and then override it accordingly:

```
//Create default Values
var defaultEventParams = {
            eventCategory : "pingData";
            eventSubCategory : "",
            eventType : "",
            eventSubType : ""
};
//Override with default values if not present
for (var e in defaultEventParams) {
    if (defaultEventParams.hasOwnProperty(e) && !eventParams(e)) {
        eventParams[e] = defaultEventParam[e];
    }
}
```